models/user.go

```diff
@@ -1078,18 +1078,52 @@ func validateUser(u *User) error {
 	return ValidateEmail(u.Email)
 }
 
-func updateUser(e db.Engine, u *User) error {
+func updateUser(ctx context.Context, u *User, emailChanged bool) error {
 	if err := validateUser(u); err != nil {
 		return err
 	}
 
+	e := db.GetEngine(ctx)
+
+	if emailChanged {
```

Comment:
Why don't we just always do this?

Reply:
It's unnecessary to always update if the primary email hasn't been changed. In fact, currently users cannot change them himself, only site admin could change them.

Reply:
Then the variable name should be `changePrimaryEmail`